### PR TITLE
PE-1004: Friday 3rd June 2022 Goerli Exec

### PR DESCRIPTION
**Pull request by brianmcmichael**
# Description

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier override
- [x] Verify expiration (`30 days` unless otherwise specified)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [x] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [x] Ensure contract is verified on `Goerli` etherscan
- [x] Change test to use Goerli spell address and deploy timestamp
- [x] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [x] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

### CODE REVIEW

**Review by talbaneth — COMMENTED** [review]
Changes look good so far, see very minor comments.

**Comment by talbaneth** [comment]
Recent changes lookg good and tests pass.
I'm not sure we should leave all the commented out mainnet stuff as I didn't see it done before and feel such thing should be a standard (either to always do it or always not do it).

**Review by talbaneth — APPROVED** [review]
There's still the comment I had for removing the mainnet stuff comments, but not feeling to strong about it.
LGTM for deployment.

**Review by talbaneth — APPROVED** [review]
Tests pass with deployed spell.
Good to cast.

File: src/Goerli-DssSpell.t.sol [hunk 1]
```diff
@@ -23,8 +24,45 @@ interface DenyProxyLike {
...
+}
+
 contract DssSpellTest is GoerliDssSpellTestBase {
 
+    function testStarknetUpdates() public {
+
+        // Test before spell
+        // Currently 1M on Goerli, 100k on mainnet
+        assertEq(
+            StarknetLike(addr.addr("STARKNET_DAI_BRIDGE")).ceiling(),
```
Comment (iamchrissmith) [inline src/Goerli-DssSpell.t.sol:38]: I'm getting a test failure here:

```haskell
Error: a == b not satisfied [uint]
    Expected: 1000000000000000000000000
      Actual: 50000000000000000000
```
at block 7016000

Comment (brianmcmichael) [reply src/Goerli-DssSpell.t.sol:38]: Yes, someone on the starknet team recently updated this. We will need to examine the mainnet bridge to see if Starknet team is retaining any authority over that contract.

https://goerli.etherscan.io/tx/0xdc31797b14d7be07158bc95415e4d5a9e95d00725a9eda2d781db4949145e61d

File: src/Goerli-DssSpell.sol [hunk 2]
```diff
@@ -47,28 +71,145 @@ contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
...
+
+        // Maximum Debt Ceiling Increase
+        //
+        //    Increase MANA-A Maximum Debt Ceiling from 10 million DAI to 15 million DAI
+        DssExecLib.setIlkAutoLineDebtCeiling("MANA-A", 15 * MILLION);
+
+        // D3M Target Borrow Rate Decrease
+        //
+        //    Reduce DIRECT-AAVEV2-DAI Target Borrow Rate from 3.5% to 2.75%
+        // MAINNET ONLY
```
Comment (iamchrissmith) [inline src/Goerli-DssSpell.sol:123]: I would probably add a comment about the D3M debt ceiling reduction and that it is only on mainnet

**Review by iamchrissmith — COMMENTED** [review]
this looks good to me once we fix that test

* [x] Standard Code review
    * [X] Check Starknet contracts against discord and etherscan
* [x] Office Hours 
    * [x] off
    * [x] matches exec doc
* [x] 30 Days Expiry
* [x] Pragma OK
* [x] Math OK
* [x] ChainLog OK
* [x] addresses_goerli.sol Matches
* [x] Spell Actions Look Good
* [x] Tests Look Good

**Review by iamchrissmith — APPROVED** [review]
changes look good to me

File: src/Goerli-DssSpell.sol [hunk 3]
```diff
@@ -24,14 +25,37 @@ import "dss-exec-lib/DssAction.sol";
...
+}
+
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
 
     // Provides a descriptive tag for bot consumption
     string public constant override description = "Goerli Spell";
 
+    address constant STARKNET_ESCROW_MOM    = 0x464379BD1aC523DdA45b7B78eCB1F703661cad2a;
+    address constant STARKNET_ESCROW        = 0x38c3DDF1eF3e045abDDEb94f4e7a1a0d5440EB44;
+    address constant STARKNET_DAI_BRIDGE    = 0xd8beAa22894Cd33F24075459cFba287a10a104E4;
```
Comment (godsflaw) [inline src/Goerli-DssSpell.sol:45]: When checking wards, it was clear this was deployed before the ESM changes.  Indeed the old ESM is made a ward on this contract, but not the new one.  We should check this in `mainnet` too.  I think we need to `deny` the old ESM ward in this spell, and add the new ESM too.

File: src/Goerli-DssSpell.sol (hunk 2 above)
Comment (brianmcmichael) [reply src/Goerli-DssSpell.sol:123]: Tracked at https://github.com/makerdao/spells-goerli/pull/78/files#diff-5ba6d6e31f0110bc3d9054811b3fd5bb9720eca165ac02046e44e3375c2f1d35R110

File: src/Goerli-DssSpell.sol [hunk 4]
```diff
@@ -24,14 +25,37 @@ import "dss-exec-lib/DssAction.sol";
...
+    function setCeiling(uint256) external;
+}
+
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
 
     // Provides a descriptive tag for bot consumption
     string public constant override description = "Goerli Spell";
 
+    address constant STARKNET_ESCROW_MOM    = 0x464379BD1aC523DdA45b7B78eCB1F703661cad2a;
+    address constant STARKNET_ESCROW        = 0x38c3DDF1eF3e045abDDEb94f4e7a1a0d5440EB44;
```
Comment (godsflaw) [inline src/Goerli-DssSpell.sol:44]: Same ESM wards comment as the STARKNET_DAI_BRIDGE.

File: src/Goerli-DssSpell.t.base.sol [hunk 5]
```diff
@@ -1117,7 +1132,9 @@ contract GoerliDssSpellTestBase is Config, DSTest, DSMath {
             uint256 ward = abi.decode(data, (uint256));
             if (ward > 0) {
                 if (skipWards(_addr, deployers.addr(i))) continue; // ONLY ON GOERLI
-                emit Log("Bad auth", deployers.addr(i), contractName);
+                emit log("Error: Bad Auth");
+                emit log_named_address(" Deployer Address: ", deployers.addr(i));
+                emit log_named_string( "Affected Contract: ", contractName);
```
Comment (godsflaw) [inline src/Goerli-DssSpell.t.base.sol:1137]:
```suggestion
                emit log_named_string("Affected Contract: ", contractName);
```

File: src/Goerli-DssSpell.sol (hunk 3 above)
Comment (brianmcmichael) [reply src/Goerli-DssSpell.sol:45]: Great catch. Pushed an update to deauth the old ESM and auth the new one.

File: src/Goerli-DssSpell.t.base.sol [hunk 6]
```diff
@@ -1117,7 +1132,9 @@ contract GoerliDssSpellTestBase is Config, DSTest, DSMath {
             uint256 ward = abi.decode(data, (uint256));
             if (ward > 0) {
                 if (skipWards(_addr, deployers.addr(i))) continue; // ONLY ON GOERLI
-                emit Log("Bad auth", deployers.addr(i), contractName);
+                emit log("Error: Bad Auth");
+                emit log_named_address(" Deployer Address: ", deployers.addr(i));
```
[the remark this answers is not in the document]
Comment (brianmcmichael) [reply src/Goerli-DssSpell.t.base.sol:1136]: This is for alignment in output. Not changing.

**Review by iamchrissmith — APPROVED** [review]
new changes lgtm

**Review by iamchrissmith — APPROVED** [review]
changes look good, tests pass:

```haskell
Running 16 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 1012862)
[PASS] testFailTooEarly() (gas: 12528)
[PASS] testFailTooLate() (gas: 12549)
[PASS] testFailWrongDay() (gas: 12572)
[PASS] testFail_notScheduled() (gas: 14139)
[PASS] testNewChainlogValues() (gas: 1062772)
[PASS] testOnTime() (gas: 1009251)
[PASS] testSpellIsCast_GENERAL() (gas: 26163537)
[PASS] testStarknetUpdates() (gas: 1064858)
[PASS] test_auth() (gas: 13204163)
[PASS] test_auth_in_sources() (gas: 4556297)
[PASS] test_bytecode_matches() (gas: 2313183)
[PASS] test_chainlog_values() (gas: 5724818)
[PASS] test_chainlog_version_bump() (gas: 2704806)
[PASS] test_nextCastTime() (gas: 341464)
[PASS] test_use_eta() (gas: 340274)
Test result: ok. 16 passed; 0 failed; finished in 674.11s
```

**Pull request merged**